flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/data/FlinkParquetWriters.java

```diff
@@ -498,6 +498,12 @@ private static class RowDataWriter extends ParquetValueWriters.StructWriter<RowD
 
     @Override
     protected Object get(RowData struct, int index) {
+      // Be sure to check for null values, even if the field is required. Without an explicit null
+      // check, BinaryRowData will ignore the null flag and parse random bytes as actual values.
+      // This will produce incorrect writes instead of failing with a NullPointerException.
+      if (struct.isNullAt(index)) {
+        return null;
+      }
```

Comment:
Actual fix is here.

Reply:
do we need to fix the `FlinkOrcWriters`?

Reply:
You are right. I revised the approach because it was too easy to miss instances. I'm instead wrapping RowData#createFieldGetter to make sure to null-check also for required / NonNull types. I'll raise an issue on the Flink side as well.

Reply:
Also see https://github.com/apache/iceberg/pull/12049#discussion_r1937962188.